sycl/source/detail/graph_impl.hpp

```diff
@@ -74,13 +77,16 @@ struct node_impl {
       if (!i->MScheduled)
         i->topology_sort(schedule);
     }
-    schedule.push_front(node_ptr(this));
+    if(MBody != nullptr)
+      schedule.push_front(node_ptr(this));
   }
 };
 
 struct graph_impl {
   std::set<node_ptr> MRoots;
   std::list<node_ptr> MSchedule;
+  // TODO: Integrate allocs into the graph
+  std::list<std::tuple<void*&,size_t,usm::alloc>> MAllocs;
```

Comment:
Preference for a `struct` rather than a `std::tuple` here. I find `foo.size` more readable than `std::get<1>(foo)`

Reply:
We certainly can use a struct here. Because of the pointer reference we need to define a constructor. Which doesn't give us any advantage when initializing the struct. Obviously self documenting when the pointer is set. I've drafted the changes, please have a look.